File: backend/utils/response.py
from pydantic import BaseModel
from typing import Optional, Any


class ApiResponse(BaseModel):
    """Unified API response format"""

    success: bool = True
    message: str = "Success"
    data: Optional[Any] = None
    error: Optional[str] = None

    @classmethod
    def ok(cls, data: Any = None, message: str = "Success"):
        """Create a success response"""
        return cls(success=True, message=message, data=data)

    @classmethod
    def fail(cls, message: str, error: str = None):
        """Create an error response"""
        return cls(success=False, message=message, error=error)

File: backend/src/utils/response.ts
export function successResponse<T>(data: T | null = null, message = 'Success') {
  return { success: true as const, message, data, error: null };
}

export function errorResponse(message: string, error: unknown = null) {
  return { success: false as const, message, data: null, error };
}
